### DISCUSSION TITLE
Source generators for MVVMToolkit not working in Visual Studio 17.5.5

### DISCUSSION THREAD

**Discussion by FLAMESpl** [CommunityToolkit/dotnet#709] [Q&A] [answered]
Hi, does anyone managed to get into working state `ObservableProperty` generators from MVVMToolkit under Visual Studio 17.5.5?

In my case when I start with blank project and start typing some view model with observable properties then sources are not generated for moreover half of fields. I try closing, opening, removing, readding bu most of the time it is not working or when one property is added then another is magically removed. The only thing that can add missing properties is removing bin and obj directories then build from scratch.

Sometimes there are compilation errors even when property is generated like on this screen:

![image](https://github.com/CommunityToolkit/dotnet/assets/17801634/f8297701-b263-4baf-9889-716fb5b8d373)

I would really like to use MVVMToolkit but I just cannot afford importing so unreliable library to company's projects.

**Comment by Sergio0694** [comment, marked as answer]
Sorry for the inconvenience! This is a known issue with Visual Studio (and Roslyn specifically). You can track it in https://github.com/CommunityToolkit/dotnet/issues/493. Could you try updating to latest Visual Studio (17.6.2) and checking again? That version should hopefully be a bit more resilient. Thank you! 🙂

**Reply by FLAMESpl** [reply]
thanks for the ticket
